src/Components/Components/src/EventCallbackFactoryBinderExtensions.cs

```diff
@@ -246,28 +260,83 @@ private static bool ConvertToNullableDecimalCore(object obj, out decimal? value)
             return true;
         }
 
-        private static class EnumConverter<T> where T : struct, Enum
+        private static BindConverter<DateTime> ConvertToDateTime = ConvertToDateTimeCore;
+        private static BindConverter<DateTime?> ConvertToNullableDateTime = ConvertToNullableDateTimeCore;
+
+        private static bool ConvertToDateTimeCore(object obj, out DateTime value)
         {
-            public static readonly BindConverter<T> Convert = ConvertCore;
+            var text = (string)obj;
+            if (string.IsNullOrEmpty(text))
+            {
+                value = default;
+                return false;
+            }
 
-            public static bool ConvertCore(object obj, out T value)
+            if (!DateTime.TryParse(text, CultureInfo.CurrentCulture, DateTimeStyles.None, out var converted))
             {
-                var text = (string)obj;
-                if (string.IsNullOrEmpty(text))
-                {
-                    value = default;
-                    return true;
-                }
+                value = default;
+                return false;
+            }
 
-                if (!Enum.TryParse<T>(text, out var converted))
-                {
-                    value = default;
-                    return false;
-                }
+            value = converted;
+            return true;
+        }
 
-                value = converted;
+        private static bool ConvertToNullableDateTimeCore(object obj, out DateTime? value)
+        {
+            var text = (string)obj;
+            if (string.IsNullOrEmpty(text))
+            {
+                value = default;
                 return true;
             }
+
+            if (!DateTime.TryParse(text, CultureInfo.CurrentCulture, DateTimeStyles.None, out var converted))
+            {
+                value = default;
+                return false;
+            }
+
+            value = converted;
+            return true;
+        }
+
+        private static bool ConvertToEnum<T>(object obj, out T value) where T : struct, Enum
+        {
+            var text = (string)obj;
+            if (string.IsNullOrEmpty(text))
+            {
+                value = default;
+                return true;
+            }
+
+            if (!Enum.TryParse<T>(text, out var converted))
+            {
+                value = default;
+                return false;
+            }
+
+            value = converted;
+            return true;
+        }
+
+        private static bool ConvertToNullableEnum<T>(object obj, out Nullable<T> value) where T : struct, Enum
```

Comment:
Added support for nullable enums without boxing. Type converters can do this, but by special casing we can avoid the boxing.